gutenberg/gutenberg.php

```diff
@@ -1,101 +1,75 @@
-    foreach($licences as $_licence_name => $_licence_data) {
-      $licenses_js_array[] = array('value' => $_licence_name, 'label' => $_licence_name);
-    }
-  }
-  wp_localize_script('source-to-core-image-block', 'isc_options', $licenses_js_array );
-  if(function_exists('wp_set_script_translations')) {
-    wp_set_script_translations('source-to-core-image-block', 'image-source-control-isc', ISCPATH . 'languages');
-  }
-}
-
-//Making our custom meta fields available to wp rest api.
-function isc_register_api_fields() {
- register_rest_field( 'attachment', 'isc_image_source',
-    array(
-      'get_callback'    => 'isc_get_attachment_meta',
-      'update_callback' => 'slug_update_attachment_meta',
-      'schema'          => null,
-      'auth_callback' => function(){
-        return current_user_can('edit_posts');
-       }
-    )
- );
- register_rest_field( 'attachment', 'isc_image_source_own',
-    array(
-       'get_callback'    => 'isc_get_attachment_meta',
-       'update_callback' => 'slug_update_attachment_meta',
-       'schema'          => null,
-       'auth_callback' => function(){
-         return current_user_can('edit_posts');
-        }
-    )
- );
- register_rest_field( 'attachment', 'isc_image_source_url',
-    array(
-       'get_callback'    => 'isc_get_attachment_meta',
-       'update_callback' => 'slug_update_attachment_meta',
-       'schema'          => null,
-       'auth_callback' => function(){
-         return current_user_can('edit_posts');
-        }
-    )
- );
- register_rest_field( 'attachment', 'isc_image_licence',
-    array(
-       'get_callback'    => 'isc_get_attachment_meta',
-       'update_callback' => 'slug_update_attachment_meta',
-       'schema'          => null,
-       'auth_callback' => function(){
-         return current_user_can('edit_posts');
-        }
-    )
- );
-}
-
-//Function to get a meta field, otherwise it will return a null value
-function isc_get_attachment_meta($object, $field_name, $request) {
-  return get_post_meta($object['id'], $field_name, true);
-}
-//Function to get a meta field, otherwise it becomes read only.
-function slug_update_attachment_meta($value, $object, $field_name ) {
-  return update_post_meta($object->ID, $field_name, $value);
+class Isc_Gutenberg{
+	
+	/**
+	 * Construct an instance of Isc_Gutenberg
+	 */
+	public function __construct() {
+		if ( !function_exists( 'register_block_type' ) ) {
+			return;
+		}
+		add_action( 'enqueue_block_editor_assets', array( $this, 'editor_assets' ) );
+		add_action( 'wp_ajax_isc_save_meta', array( $this, 'save_meta' ) );
+	}
+	
+	public function save_meta() {
+		$_post = wp_unslash( $_POST );
+		if ( isset( $_post['nonce'] ) && false !== wp_verify_nonce( $_post['nonce'], 'isc-gutenberg-nonce' ) ) {
+			if ( current_user_can( 'edit_posts' ) ) {
+				update_post_meta( absint( $_post['id'] ), $_post['key'], $_post['value'] );
+				wp_send_json( $_post );
+			}
+		}
+		die;
+	}
+	
+	/**
+	 * Enqueue JS file
```

Comment:
This function seems to do more than just enqueuing JS files. Please leave more information in the description and as inline comments

Reply:
https://github.com/cnhk/image-source-control/commit/edf48b25d8aa3a25c54df840da03d3a2b06afc81